Demo/DemoView.swift

```diff
@@ -0,0 +1,94 @@
+//  GpxLocationManager
+//
+//  Created by Joshua Adams on 4/14/18.
+//  Copyright © 2018 Josh Adams. All rights reserved.
+//
+
+import UIKit
+import MapKit
+
+class DemoView: UIView {
+  internal let mapView: MKMapView = {
+    let mapView = MKMapView()
+    return mapView
+  } ()
+
+  internal let gpxControl: UISegmentedControl = {
+    let gpxControl = UISegmentedControl(items: ["GPX File", "Locations", "Device GPS"])
+    gpxControl.selectedSegmentIndex = 0
+    return gpxControl
+  } ()
+
+  internal let speedStepper: UIStepper = {
+    let speedStepper = UIStepper()
+    return speedStepper
+  } ()
+
+  private let speedLabelLabel: UILabel = {
+    let speedLabelLabel = UILabel()
+    speedLabelLabel.text = "Speed: "
+    speedLabelLabel.textColor = UIColor.white
+    return speedLabelLabel
+  } ()
+
+  internal let speedLabel: UILabel = {
+    let speedLabel = UILabel()
+    speedLabel.text = " "
+    speedLabel.textColor = UIColor.white
+    return speedLabel
+  } ()
+
+  private let standard: CGFloat = 8.0
+  private let disabledAlpha: CGFloat = 0.4
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    [mapView, gpxControl, speedLabelLabel, speedLabel, speedStepper].forEach {
+      guard let control = $0 as? UIView else { fatalError("Could not cast \($0) to UIView.") }
+      control.enableAutoLayout()
+      addSubview(control)
+    }
+
+    mapView.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor).activate()
+    mapView.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor).activate()
+    mapView.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor).activate()
+    mapView.bottomAnchor.constraint(equalTo: gpxControl.topAnchor, constant: standard * -1.0).activate()
+
+    gpxControl.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor).activate()
+    gpxControl.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor).activate()
+    gpxControl.bottomAnchor.constraint(equalTo: speedStepper.topAnchor, constant: standard * -1.0).activate()
+
+    speedStepper.leadingAnchor.constraint(equalTo: speedLabel.trailingAnchor, constant: standard).activate()
+    speedStepper.bottomAnchor.constraint(equalTo: layoutMarginsGuide.bottomAnchor, constant: standard * -1.0).activate()
+
+    speedLabel.centerXAnchor.constraint(equalTo: centerXAnchor).activate()
+    speedLabel.centerYAnchor.constraint(equalTo: speedStepper.centerYAnchor).activate()
+
+    speedLabelLabel.trailingAnchor.constraint(equalTo: speedLabel.leadingAnchor, constant: standard * -1.0).activate()
+    speedLabelLabel.centerYAnchor.constraint(equalTo: speedStepper.centerYAnchor).activate()
+  }
+
+  required init(coder aDecoder: NSCoder) {
+    fatalError("This class does not support NSCoding.")
+  }
+
+  internal func updateSpeedLabel(speed: Double) {
+    speedLabel.text = "\(Int(speed))X"
+  }
+
+  internal func enableSpeedControls() {
+    [speedLabelLabel, speedLabel, speedStepper].forEach {
```

Comment:
These should be class variables so we don't have copy and paste code

Reply:
Disregard, I pulled the files and see what you are doing. I'm good with the way it is :)